### PR TITLE
IA-512 make errorCode nullable

### PR DESCRIPTION
**Pull request by Qi77Qi**
https://broadworkbench.atlassian.net/browse/IA-512

There have been too many times the thought "this thing really should be nullable" crossed my mind, hence decided to just take a few min to make it happen...

Searched through jira, and realize there was a ticket filed for this

---
Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; Delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: http/src/main/resources/org/broadinstitute/dsde/workbench/leonardo/liquibase/changesets/20201026_drop_not_null_constraint_on_error_code_table.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<databaseChangeLog logicalFilePath="leonardo" xmlns="http://www.liquibase.org/xml/ns/dbchangelog" xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog-ext http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.4.xsd">
+    <changeSet logicalFilePath="leonardo" author="qi" id="drop_on_update_pd_table">
```
Comment (rtitle) [inline http/src/main/resources/org/broadinstitute/dsde/workbench/leonardo/liquibase/changesets/20201026_drop_not_null_constraint_on_error_code_table.xml]: Change `id` field?

**Comment by rtitle** [comment]
Looks good, 1 comment in the migration

File: http/src/main/resources/org/broadinstitute/dsde/workbench/leonardo/liquibase/changesets/20201026_drop_not_null_constraint_on_error_code_table.xml (hunk 1 above)
Comment (Qi77Qi) [reply http/src/main/resources/org/broadinstitute/dsde/workbench/leonardo/liquibase/changesets/20201026_drop_not_null_constraint_on_error_code_table.xml]: updated!

**Review by rtitle — APPROVED** [review]

**Review by kyuksel — APPROVED** [review]
:shipit:

**Pull request merged**